rpc/src/cli/cli.rs

```diff
@@ -9,20 +10,37 @@ pub struct Cli {
     #[arg(short, long, value_name = "FILE", default_value = config::CONFIG_FILENAME)]
     pub config: Option<PathBuf>,
 
+    #[arg(short, long)]
+    /// Binding port number
+    port: Option<u16>,
+
+    // short option `-h` conflicts with `--help`, so it is not enabled.
+    #[arg(long)]
+    /// Binding host
+    host: Option<String>,
+
     #[command(subcommand)]
     pub command: Command,
 }
 
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Subcommand)]
 pub enum Command {
-    Serve {
-        #[arg(short, long)]
-        /// Binding port number
-        port: Option<u16>,
-
-        // short option `-h` conflicts with `--help`, so it is not enabled.
-        #[arg(long)]
-        /// Binding host
-        host: Option<String>,
+    Serve,
+
+    #[command(subcommand)]
+    Request(Request),
+}
+
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Subcommand)]
+pub enum Request {
+    CheckNullifiers {
+        #[arg(value_parser=parse_nullifier)]
+        /// List of nullifiers to check
+        nullifiers: Vec<String>,
     },
 }
+
+fn parse_nullifier(value: &str) -> Result<Digest, String> {
+    let data = hex::decode_hex(str);
+    if data.len() !=
+}
```

Comment:
This is probably a partial commit.

Reply:
oh! oops, removed. Thanks for flagging it!